app/controllers/posts_controller.rb

```diff
@@ -139,10 +143,19 @@ def update
 
     @post.assign_attributes(post_params)
     @post.board ||= Board.find(3)
-
-    preview and return if params[:button_preview].present?
-
-    @post.build_new_tags_with(current_user)
+    settings = process_tags(Setting, :post, :setting_ids)
+    warnings = process_tags(ContentWarning, :post, :content_warning_ids)
+    labels = process_tags(Label, :post, :label_ids)
+
+    if params[:button_preview].present?
+      @post.association(:settings).target = []
+      @post.association(:content_warnings).target = []
+      @post.association(:labels).target = []
+      settings.each { |s| @post.association(:settings).add_to_target(s) }
+      warnings.each { |w| @post.association(:content_warnings).add_to_target(w) }
+      labels.each { |l| @post.association(:labels).add_to_target(l) }
```

Comment:
What's this bit working around?

Reply:
So that they show properly in preview! this way they are in memory, whereas using @post.<name> = thing will actually create the records in the database, which is not at all the goal of preview.

Reply:
Ahhh. Okay. I didn't know where to find documentation on these methods.